api/include/opentelemetry/nostd/pair.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "opentelemetry/nostd/utility.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace nostd
+{
+
+// The nostd::pair class provides a data structure capable of hold
+// two values of different or the same types, and provides access to each.
+
+template <class T1, class T2>
+class pair
+{
+public:
+  pair()
+  {
+    first_  = T1();
+    second_ = T2();
+  }
```

Comment:
Please use initialization list. Also, it will fail if T1 and T2 doesn't have default constructor ( which is acceptable I believe).

Reply:
Removed T1 and T2 default constructors.